core/transactions.hxx

```diff
@@ -127,21 +129,49 @@ using txn_complete_callback =
 class transactions : public couchbase::transactions::transactions
 {
   public:
+    /**
+     * @internal
+     * Called internally
+     */
+    transactions(core::cluster cluster, const couchbase::transactions::transactions_config::built& config);
```

Comment:
Should I make the constructor private?

Reply:
yes, it makes sense. Everything will go through `create` method then.

Reply:
I agree -- I think with the changes, we need to force the use of the `transactions::create()`.  I suppose it is not a problem if not using the metadata collection, but I think going the route of not providing any "foot-guns" is the better path.

Reply:
It appears that you can't use `std::make_shared<transactions>` if the constructor is private, and the workarounds are slightly ugly. I think I'm leaning towards leaving them public as it's a core API, though happy to do otherwise.